### PR TITLE
Create enum to record reasons for marking GRF frame as "missing" // writeCSVData fix

### PR DESCRIPTION
**Pull request by nickbianco**
This PR adds the new enum `MissingGRFReason`, which will let us keep track of why a frame was marked as "possiblyMissingGRF". I think I covered all of the possible cases, but feel free to point out any other that I might have missed. 

I mostly need feedback on the changes to `SubjectOnDisk`, some of that code was a little unclear to me. Specifically, should I make any additions here?

https://github.com/keenon/nimblephysics/blob/8ebc50ade44a0a659e1f945334b9bdc3a2b9c71d/stubs/_nimblephysics-stubs/biomechanics/__init__.pyi#L326

There is also a small fix in `writeCSVData()`, where I messed up printing out the time column.

### CODE REVIEW

File: dart/biomechanics/DynamicsFitter.cpp [hunk 1]
```diff
@@ -10143,6 +10153,7 @@ void DynamicsFitter::estimateUnmeasuredExternalForces(
                   << trial << " at time " << t << std::endl;
         filteredTimesteps.push_back(t + 1);
         init->probablyMissingGRF[trial][t + 1] = true;
+        init->missingGRFReason[trial][t + 1] = MissingGRFReason::unmeasuredForce;
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10156]: Maybe we rename this to `unmeasuredExternalForceDetected` or something, to make it a bit clearer what's happening?

File: dart/biomechanics/DynamicsFitter.cpp [hunk 2]
```diff
@@ -10153,6 +10164,7 @@ void DynamicsFitter::estimateUnmeasuredExternalForces(
                     << std::endl;
           filteredTimesteps.push_back(t + 1);
           init->probablyMissingGRF[trial][t + 1] = true;
+          init->missingGRFReason[trial][t + 1] = MissingGRFReason::missingGRF;
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10167]: Maybe this gets named `measuredGrfZeroWhenAccelerationNonZero`?

File: dart/biomechanics/DynamicsFitter.cpp [hunk 3]
```diff
@@ -10253,6 +10267,7 @@ int DynamicsFitter::estimateUnmeasuredExternalTorques(
         if (init->probablyMissingGRF.size() > trial)
         {
           init->probablyMissingGRF[trial][t] = true;
+          init->missingGRFReason[trial][t] = MissingGRFReason::unmeasuredTorque;
```
Comment (keenon) [inline dart/biomechanics/DynamicsFitter.cpp:10270]: Maybe for consistency with `forceDiscrepancy`, we call this `torqueDiscrepancy`?

File: python/_nimblephysics/biomechanics/SubjectOnDisk.cpp [hunk 4]
```diff
@@ -205,6 +206,14 @@ Note that these are specified in the local body frame, acting on the body at its
 
             This method is provided to give a cheaper way to filter out frames we want to ignore for training, without having to call
             the more expensive :code:`loadFrames()`
+          )doc")
```
Comment (keenon) [inline python/_nimblephysics/biomechanics/SubjectOnDisk.cpp:209]: I think you also need to add an explicit PyBind11 enum declaration for your new binding, if you don't want it to blow up when you return from this. Docs here: https://pybind11.readthedocs.io/en/stable/classes.html#enumerations-and-internal-types

**Review by keenon — COMMENTED** [review]
Looks lovely, just some minor naming requests, and needing to add the enum binding for Python

File: python/_nimblephysics/biomechanics/SubjectOnDisk.cpp (hunk 4 above)
Comment (nickbianco) [reply python/_nimblephysics/biomechanics/SubjectOnDisk.cpp:209]: Okay, I attempted wrapping the enum, but I'm not exactly sure the best way to test it.

Comment (nickbianco) [reply python/_nimblephysics/biomechanics/SubjectOnDisk.cpp:234]: Okay, should work now. The pybind stub regeneration worked.

**Review by keenon — APPROVED** [review]

**Pull request merged**